src/fondant/core/schemas/common.json

```diff
@@ -51,15 +51,30 @@
           ]
         }
       },
-      "required": ["type"],
+      "required": [
+        "type"
+      ],
       "additionalProperties": false
     },
+    "additionalFields": {
```

Comment:
do you mean adding `additionalFields` within the `additionalProperties` of `field` or simply enabling `additionalProperties`  to be set to true in the component spec to define generic components?

Reply:
> simply enabling additionalProperties to be set to true in the component spec to define generic components?

This one.

So using `additionalProperties` instead of `additionalFields`.

Reply:
Yes that would be fine for me.  Json schema itself is defined by a meta schema, which does the same thing:

```json
"additionalProperties": {
    "anyOf": [
        { "type": "boolean" },
        { "$ref": "#" }
    ],
    "default": {}
},

I would maybe change the default to match the meta schema, it should still evaluate to False:
```json
"additionalProperties": {
  "type": "boolean",
  "default": {},
},
```

Then we can easily extend it in the future.